### ISSUE TITLE
SWY is_close behaves incorrectly with infinite nodata values

### ISSUE THREAD

**Issue by emlys** [natcap/invest#331]
Came across this while looking into [this forum question](https://community.naturalcapitalproject.org/t/problem-with-baseflow-and-quickflow-estimates/1394).
In seasonal_water_yield_core.pyx, the function:
```
cdef int is_close(double x, double y):

    return abs(x-y) <= (1e-8+1e-05*abs(y))
```
doesn't work as intended when one of the arguments is infinite. This happens when a user inputs monthly Et0 or precipitation rasters with an infinite nodata value. `is_close(x, -inf) == True` for many (all?) normal values of x, resulting in many pixels not being included in the yearly totals.

**Comment by dcdenu4** [comment]
What does a nodata value look like that returns `-inf`?

**Comment by emlys** [comment]
`pygeoprocessing.get_raster_info` returns -1.797693e+308 for the nodata value, but I guess something goes wrong when it gets squeezed into the variable `cdef float precip_nodata`

**Comment by phargogh** [comment]
> nothing's stopping someone from using an even more gigantic value...

True! Though it also looks like if someone wants to exceed the limits of a `double`, the GDAL devs will need to support that (they assume a gtiff nodata will fit into a double): https://github.com/OSGeo/gdal/blob/master/gdal/frmts/gtiff/geotiff.cpp#L634

**Comment by dcdenu4** [comment]
Fixed in PR #343
